lib/error.g

```diff
@@ -59,16 +59,58 @@ end);
 
 ErrorLVars := fail;
 
-BIND_GLOBAL("WHERE", function( context, depth, outercontext)
-    local   bottom,  lastcontext,  f;
+BIND_GLOBAL("PRETTY_PRINT_VARS", function(context)
+    local vars, i, argcount, val;
+    vars := ContentsLVars(context);
+    if IsRecord(vars) then
+        argcount := ABS_RAT(NumberArgumentsFunction(vars.func));
+        PrintTo(ERROR_OUTPUT, "\>\>\n\<\<arguments:");
```

Comment:
That doesn't make sense to me. Why would you decrease the hint level with `\>\>` at the start, and then after the newline increase it again? Is that to counteract some line break hints in the caller? But that would seem rather fragile, and at least should be documented.

Reply:
So, \>\> increases the hint level (at least, it causes an indent). The reason I write this strangely is because (as I mention below, but I should pull the comment up) there is a long-standing bug in `PrintTo(ERROR_OUTPUT,` (and a related places in printing) where the indentation level gets reset on each print call.

Therefore I'm making sure I do all the indentation I want in a single call to PrintTo is balanced (which I believe I've achieved, unless I'm missing something), so that at the moment I end up with the output I want, and if PrintTo is ever fixed, the output won't change.

Reply:
I'll write a longer comment pointing this out at the top of the function

Reply:
Thanks for the explanation, I mixed up the meaning of `\<` and `\>`.

And: Ah yeah, I remember that bug: the problem is that for each such `PrintTo` call, we "open" stderr, print to it, and close it -- but the output state for the stdout and stderr stream are not shared, but arguably should be, at least if both are known to map to the same device. We probably should have an issue for that (perhaps we already do, I didn't check).